src/main/java/com/conveyal/gtfs/graphql/GTFSGraphQL.java

```diff
@@ -1,41 +1,93 @@
 package com.conveyal.gtfs.graphql;
 
+import com.conveyal.gtfs.graphql.fetchers.JDBCFetcher;
+import graphql.ExecutionInput;
 import graphql.GraphQL;
+import graphql.execution.instrumentation.dataloader.DataLoaderDispatcherInstrumentation;
+import graphql.introspection.IntrospectionQuery;
+import graphql.schema.DataFetchingEnvironment;
+import org.dataloader.DataLoader;
+import org.dataloader.DataLoaderRegistry;
 
 import javax.sql.DataSource;
-import java.sql.Connection;
-import java.sql.SQLException;
-
-/**
- * This provides a GraphQL API around the gtfs-lib JDBC storage.
- * This just makes a Java API with the right schema available, which uses String requests and responses.
- * To make this into a web API you need to wrap it in an HTTP framework / server.
- */
+import java.util.List;
+import java.util.Map;
+
+import static com.conveyal.gtfs.graphql.fetchers.JDBCFetcher.newJdbcDataLoader;
+
 public class GTFSGraphQL {
 
-    private static DataSource dataSource;
+    /**
+     * Execute a graphQL request.  This method creates a new graphQL runner that is able to cache and batch sql queries
+     * during the course of obtaining the data according to the graphQL query.
+     *
+     * @param dataSource The dataSource to use in connecting to a database to make sql queries.
+     * @param query The graphQL query
+     * @param variables The variables to apply to the graphQL query
+     * @return
+     */
+    public static Map<String, Object> run (DataSource dataSource, String query, Map<String,Object> variables) {
+        // the registry and dataLoaders must be created upon each request to avoid caching loaded data after each request
+        DataLoaderRegistry registry = new DataLoaderRegistry();
+        DataLoader<JDBCFetcher.JdbcQuery, List<Map<String, Object>>> jdbcQueryDataLoader = newJdbcDataLoader();
+        registry.register("jdbcfetcher", jdbcQueryDataLoader);
+
+        return GraphQL.newGraphQL(GraphQLGtfsSchema.feedBasedSchema)
+            // this instrumentation implementation will dispatch all the dataloaders as each level fo the graphql query
+            // is executed and hence make batched objects available to the query and the associated DataFetchers.  This
+            // must be created new each time to avoid caching load results in future requests.
```

Comment:
The Javadoc for the method mentions that we want to cache results, then here it says that we don't want to "cache load results in future requests". It's not clear to me what this means. Do you mean that we want a fresh, empty cache for each request, i.e. that we don't want cached results to be reused by future (or concurrent) requests? Is this because we expect the fetched values to change between requests? Please clarify in the comments/javadoc.